configure

```diff
@@ -808,10 +810,34 @@ $opts{'boost'} = '' if exists $opts{'boost-version'} && !exists $opts{'boost'};
 my %use_system_pkg = map {$_, 1} qw/ant glib qt boost/;
 my %use_win_default = ('boost' => 'c:\\Boost');
 
+
+sub default_java_home {
+  my $host = $opts{'host'};
+  my $result = '';
+  if ($host eq 'macosx') {
+     $result = `/usr/libexec/java_home`;
+  }
+  elsif ($host ne 'win32') {
+     $result = `readlink -f /usr/bin/javac | sed "s:/bin/javac::"`;
+  }
+  
+  chomp $result;
+  return $result;
+}
+
 for my $key (keys %optdep) {
   if (exists $opts{$key} && $opts{$key} eq '') {
     if (defined $optdep{$key}->[0] && $ENV{$optdep{$key}->[0]}) {
       $opts{$key} = $ENV{$optdep{$key}->[0]};
+      
+      if ($key eq 'java') {
+        ## when the environmental variable JAVA_HOME is set to a JRE location, 
+        ## try to resolve JAVA_HOME based on the location of javac
+        my $java_home = default_java_home();
+        if (!-r $targetEnv{$optdep{$key}->[1]} . '/' . $optdep{$key}->[2] and $java_home ne '') {
```

Comment:
This doesn't look right, doesn't "$optdep{'java'}->[1]" evaluate to "include/jni.h"?